modules/auxiliary/scanner/scada/modbusclient.rb

```diff
@@ -369,6 +386,31 @@ def write_registers
       print_error("Unknown answer")
     end
   end
+  
+  def read_id
+    @function_code = 0x2b
+    print_status("Sending READ ID...")
+    response = send_frame(make_read_id_payload)
+    if response.nil?
+      print_error("No answer for the READ REGISTER")
+    elsif response.unpack("C*")[7] == (0x80 | @function_code)
+      handle_error(response)
+    elsif response.unpack("C*")[7] == @function_code
+      @ObjCnt = response[13].unpack("C")[0]
+      @ObjIdPos=14
+      begin
+    @ObjLen = response[@ObjIdPos+1].unpack("C")[0]
```

Comment:
This all seems odd to me as your assuming the response is of a certain length. Should we not be checking to make sure the response is at least of a certain length prior to trying to parse it like this? This would prevent Out-Of-Bounds access errors.

Reply:
I see but isn't it the same for all other actions in that module? Nevertheless,

Reply:
True, however just cause it wasn't done properly in the past doesn't mean that we should continue doing it incorrectly in the future 😄 Keep in mind this module in particular is pretty old so a lot of the code was committed before we started to tighten up code quality requirements.